src/states/infinitepeps.jl

```diff
@@ -40,7 +36,9 @@ Allow users to pass in arrays of spaces.
 function InfinitePEPS(
     Pspaces::AbstractArray{S,2},
     Nspaces::AbstractArray{S,2},
-    Espaces::AbstractArray{S,2}=Nspaces,
+    Espaces::AbstractArray{S,2}=Nspaces;
+    finit=rand,
+    dtype=ComplexF64,
```

Comment:
In principle you can't really have optional arguments in the beginning, but you can spoof it:
```julia
InfinitePEPS(Pspaces, Nspaces, Espaces) = InfinitePEPS(randn, ComplexF64, Pspaces, Nspaces, Espaces)
function InfinitePEPS(f, T, Pspaces, Nspaces, Espaces)
#body
end
```

As long as you can specify the argument types well enough such that there are no ambiguities, this works.

The main reason I would put them in the beginning is because that is how TensorKit and MPSKit organizes this, which is loosely inspired on `Array{T}(undef,  size)`. I don't think it matters too much, which is why I would stick to the same format ;)

Reply:
Alright, sticking to the convention does make sense. I changed it now :)